presto-main/src/main/java/com/facebook/presto/execution/StateMachine.java

```diff
@@ -276,34 +278,22 @@ public void addStateChangeListener(StateChangeListener<T> stateChangeListener)
     /**
      * Wait for the state to not be {@code .equals()} to the specified current state.
      */
-    public Duration waitForStateChange(T currentState, Duration maxWait)
+    @VisibleForTesting
```

Comment:
Good, but as with `QueryStateMachine` -- it's better to move this logic to tests altogether.